src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -42,10 +42,16 @@ class AsyncLogWriter::AsyncLogLocker : public StackObj {
   }
 
   ~AsyncLogLocker() {
-    assert(_holder == Thread::current_or_null(), "must be");
+    assert(_holder == nullptr || _holder == Thread::current_or_null(), "must be");
```

Comment:
Don't understand this change. We are in the destructor so if the holder is null then this must be an unattached thread. So the original assert suffices.

Reply:
True, this is an unnecessary change.